packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,251 @@
 // found in the LICENSE file.
 
 import 'dart:math' as math;
+import 'dart:ui' as ui;
 
 import 'basic_types.dart';
-import 'border_radius.dart';
 import 'borders.dart';
 import 'edge_insets.dart';
 
-/// A rectangular border with smooth continuous transitions between the straight
-/// sides and the rounded corners.
+/// A rectangle border with continuous corners.
+///
+/// A shape similar to a rounded rectangle, but with a smoother transition from
+/// the sides to the rounded corners.
+///
+/// The rendered shape roughly approximates that of a superellipse. In this
+/// shape, the curvature of each 90º corner around the length of the arc is
+/// approximately a gaussian curve instead of a step function as with a
+/// traditional quarter circle round. The rendered rectangle is roughly a
+/// superellipse with an n value of 5.
+///
+/// In an attempt to keep the shape of the rectangle the same regardless of its
+/// dimension (and to avoid clipping of the shape), the radius will
+/// automatically be lessened if its width or height is less than ~3x the
+/// declared radius. The new resulting radius will always be maximal in respect
+/// to the dimensions of the given rectangle.
+///
+/// To achieve the corner smoothness with this shape, ~50% more pixels in each
+/// dimensions are used to render each corner. As such, the ~3 represents twice
+/// the ratio (ie. ~3/2) of a corner's declared radius and the actual height and
+/// width of pixels that are used to construct it. For example, if a
+/// rectangle had and a corner radius of 25, in reality ~38 pixels in each
+/// dimension would be used to construct a corner and so ~76px x ~76px would be
+/// the minimal size needed to render this rectangle.
```

Comment:
"... needed to render this rectangle before reducing the radius."

Otherwise, it might sound like we can't render this rectangle at all.